apps/aecore/src/pow_sha256.erl

```diff
@@ -0,0 +1,106 @@
+%%%=============================================================================
+%%% @copyright 2017, Aeternity Anstalt
+%%% @doc
+%%%    Module dealing with SHA256 Proof of Work calculation and hash generation
+%%% @end
+%%%=============================================================================
+-module(pow_sha256).
+
+-export([generate/3,
+         recalculate_difficulty/3]).
+
+-ifdef(TEST).
+-compile(export_all).
+-endif.
+
+-include("sha256.hrl").
+
+-define(NONCE_RANGE, 1000000000000000000000000).
+
+%% Integers encrypted in scientific notation:
+%%   256 * <base-2 exponent> + the most significant byte (i.e., the significand)
```

Comment:
I understand this is not meant for representing any arbitrary integers, please explicate constraints that I understand are:
* Non-negative integer (otherwise, is the sign represented consuming one bit of the significand? How?);
* Positive integer (otherwise could zero have multiple representations i.e. with arbitrary exponent?);
* Integer with at most 8 significant bits (or add comment that some get lost).

Also:
* Please add note that the significand can assume value from 1 to 255 (so that it is more clear that the `256*` with the exponent is for freeing one byte for the significand).